roachtest: db-console/mixed-version-endpoints failed

roachtest.db-console/mixed-version-endpoints [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/20141905?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/20141905?buildTab=artifacts#/db-console/mixed-version-endpoints) on master @ [9ddf582e7c38e1b5e844b1f6dfc014ac5f8843c0](https://github.com/cockroachdb/cockroach/commits/9ddf582e7c38e1b5e844b1f6dfc014ac5f8843c0):


```
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
failed to verify response: unexpected status code: 500, body: {
  "error": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "code": 13,
  "message": "An internal server error has occurred. Please check your CockroachDB logs for more details.",
  "details": [
  ]
}
test artifacts and logs in: /artifacts/db-console/mixed-version-endpoints/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=aws</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>mvtDeploymentMode=system-only</code>
 - <code>mvtVersions=v25.2.2 → master</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for aws clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #148875 roachtest: db-console/mixed-version-endpoints failed [C-test-failure O-roachtest O-robot T-observability branch-release-25.3]
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*db-console/mixed-version-endpoints.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

One of these `500` failure modes is when hitting the `_status/logs` endpoint:

```
[mixed-version-test/38_run-test-db-console-endpoints] 2025/08/06 08:58:09 runner.go:350: mixed-version test failure while running step 38 (run "test db console endpoints"): failed testing endpoint /_status/logs/{node_id} with other node ID: failed to verify response: unexpected status code: 500, body: {
```

In node 2's logs we see

```
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228  decoding on line 8: malformed log entry
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +(1) attached stack trace
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  -- stack trace:
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/util/log.(*entryDecoderV2).Decode.func1
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/util/log/format_crdb_v2.go:507
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/util/log.(*entryDecoderV2).Decode
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/util/log/format_crdb_v2.go:522
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/util/log.readAllEntriesFromFile
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/util/log/file_api.go:428
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/util/log.FetchEntriesFromFilesWithFormat
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/util/log/file_api.go:318
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/util/log.FetchEntriesFromFiles
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/util/log/file_api.go:295
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/server.(*statusServer).Logs
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	pkg/server/status.go:1527
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_Logs_Handler.func1
E250806 08:57:49.944051 7852 server/status.go:1530 â‹® [T3,Vmixed-version-tenant-gc8fu,client-tenant=â€¹3â€º,n2,status,tenant=3] 228 +  | 	bazel-out/k8-opt/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:9951
```

In one of the logs on node 2, we see a panic on line 8 that is presumably the malformed log entry.

```
I250806 08:52:36.997159 1 util/log/file_sync_buffer.go:237 â‹® [config]   file created at: 2025/08/06 08:52:36
I250806 08:52:36.997186 1 util/log/file_sync_buffer.go:237 â‹® [config]   running on machine: â€¹teamcity-20301899-1754458930-64-n5cpu4-0002â€º
I250806 08:52:36.997200 1 util/log/file_sync_buffer.go:237 â‹® [config]   binary: CockroachDB CCL v25.4.0-alpha.00000000-dev-ed79da5aca075bb66bf804192c4820125865b598 (x86_64-pc-linux-gnu, built 2025/08/06 05:11:56, go1.24.5 X:nocoverageredesign)
I250806 08:52:36.997205 1 util/log/file_sync_buffer.go:237 â‹® [config]   arguments: [â€¹././cockroachâ€º â€¹startâ€º â€¹--certs-dirâ€º â€¹certsâ€º â€¹--logâ€º â€¹file-defaults: {dir: 'logs', exit-on-error: false}â€º â€¹--listen-addr=:26257â€º â€¹--http-addr=:26258â€º â€¹--advertise-addr=10.142.2.178:26257â€º â€¹--join=34.74.173.195:26257â€º â€¹--storeâ€º â€¹path=/mnt/data1/cockroach,attrs=store1:node2:node2store1â€º â€¹--wal-failover=among-storesâ€º â€¹--cache=25%â€º â€¹--locality=cloud=gce,region=us-east1,zone=us-east1-bâ€º â€¹--max-sql-memory=25%â€º]
I250806 08:52:36.997221 1 util/log/file_sync_buffer.go:237 â‹® [config]   log format (utf8=âœ“): crdb-v2
I250806 08:52:36.997225 1 util/log/file_sync_buffer.go:237 â‹® [config]   line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid [chan@]file:line redactionmark \[tags\] [counter] msg
I250806 08:52:36.996885 1 util/log/flags.go:239  [-] 1  stderr capture started
panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0142f34a0 0xc0120e7740 {{} 1} {[] {} 0xc014aca180} {[] {} <nil>} {{{{} {0 0}}} 0xc014b69140}})]})

goroutine 26775 gp=0xc0174b9a40 m=9 mp=0xc001396008 [running]:
panic({0x6f77260?, 0xc001aca820?})
```

--- a/pkg/util/log/file_api.go
+++ b/pkg/util/log/file_api.go
@@ -429,6 +429,13 @@ func readAllEntriesFromFile(
 			if err == io.EOF {
 				break
 			}
+
+			// skip the malformed log entry. This is primarily observed in the
+			// panic which is unstructured.
+			// See: https://github.com/cockroachdb/cockroach/issues/151493
+			if errors.Is(err, ErrMalformedLogEntry) {
+				continue
+			}
 			return nil, false, err
 		}
 		var match bool

